packages/common/src/directives/ng_switch.ts

```diff
@@ -133,7 +133,13 @@ export class NgSwitch {
 
   /** @internal */
   _matchCase(value: any): boolean {
-    const matched = value == this._ngSwitch;
+    const matched = value === this._ngSwitch;
+    if ((typeof ngDevMode === 'undefined' || ngDevMode) && matched !== (value == this._ngSwitch)) {
+      console.warn(formatRuntimeError(
+          RuntimeErrorCode.EQUALITY_NG_SWITCH_DIFFERENCE,
+          'as of Angular v17 the NgSwitch directive uses strict equality comparison === instead of == to match different cases. ' +
```

Comment:
Quick proposal for the warning message (also to include the values, so that it's easier to locate affected ngSwitch):
```suggestion
          'As of Angular v17 the `NgSwitch` directive uses strict equality comparison `===` instead of `==` to match different cases. ' +
          `Previously, the \`${this._ngSwitch}\` switch value matched the \`${value}\` case value, which is no longer ` +
          'the case with strict equality check. ' +
          'To fix this, adjust your `ngSwitch` expression and / or `ngSwitchCase` values to work with strict equality checks.'
```

Reply:
Looking at the tests, we may need to wrap the values into quotes if it's a string to avoid showing "`1` doesn't match `1`" in the error :)
```
const ngSwitchValue = typeof this._ngSwitch === 'string' ? `"${this._ngSwitch}"` :  this._ngSwitch;
```

Reply:
Went with a variant of it, thnx!